src/ConnectedMode/Embedded/SonarLintTargets.xml

```diff
@@ -20,11 +20,14 @@
 
     <PropertyGroup Condition="$(BuildingInsideVisualStudio) == 'true'">
         <SolutionDirAsInVs>$([System.IO.Path]::GetDirectoryName($(SolutionDir)))</SolutionDirAsInVs>
+        <SonarLanguage>$(Language)</SonarLanguage>
+        <SonarLanguage Condition=" $(Language) == 'C#' OR $(Language) == 'c#'">csharp</SonarLanguage>
+        <SonarLanguage Condition=" $(Language) == 'VISUALBASIC'">vb</SonarLanguage>
         <!-- C#/VB.NET settings need to be in different folders -->
-        <_SLVSRootFolder>$(APPDATA)\SonarLint for Visual Studio\Bindings\$(SolutionName)_$(SolutionDirAsInVs.GetHashCode())\$(Language)</_SLVSRootFolder>
+        <_SLVSRootFolder>$(APPDATA)\SonarLint for Visual Studio\Bindings\$(SolutionName)_$(SolutionDirAsInVs.GetHashCode())</_SLVSRootFolder>
 
-        <_SLVSGeneratedRoslynConfigFile>$(_SLVSRootFolder)\slvs_generated.globalconfig</_SLVSGeneratedRoslynConfigFile>
-        <_SLVSGeneratedAdditionalFile>$(_SLVSRootFolder)\SonarLint.xml</_SLVSGeneratedAdditionalFile>
+        <_SLVSGeneratedRoslynConfigFile>$(_SLVSRootFolder)\$(SonarLanguage).globalconfig</_SLVSGeneratedRoslynConfigFile>
```

Comment:
I thought about placing the file under the csharp/vb folders, next to SonarLint.xml, but it wouldn't be consistent with the other setting files. I don't have a preference either way, LMK what you think would be better/simpler.

Reply:
Hm I can see where you are coming from, but to me its more confusing that we have a folder called csharp and then a file for csharp outside of said folder but that could just be a personal preference

Reply:
I'm in two minds about which I prefer. Having both settings files in the same folder feels simpler, rather than having to look in two places, and the Roslyn configuration is already inconsistent with the other languages since it contains the extra `SonarLint.xml` file.

However, the DotNet language team have looked at whether it would be possible to provide the rule configuration parameters in an `.editorconfig`/`.globalconfig` file in the future i.e. get rid of the `SonarLint.xml` file completely.  If they did that, we could get rid of the `csharp` and `vb` folders since we only need them for the `SonarLint.xml` file (this isn't something they are going to do immediately, but might in the future).

For me, there isn't much in it. Since we're not expecting users to have to look at these files directly, I would probably leave it as it is (i.e. only the `SonarLint.xml` file in a separate folder), but I don't have a strong preference for it.